src/pages/Overview/Overview.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { useEffect, useMemo } from "react"
+import { utils } from "ethers"
+import { DateTime } from "luxon"
+import { Box } from "../../components/Box"
+import { Column, Row } from "../../components/Layout"
+import { Title } from "../../components/Title"
+import { Chart } from "./Chart"
+
+import { commify } from "../../utils/units"
+import { useTVLOverTime } from "../../hooks/api/useTVLOverTime"
+import { useTVCOverTime } from "../../hooks/api/useTVCOverTime"
+
+import styles from "./Overview.module.scss"
+
+type ChartDataPoint = {
+  name: string
+  value: number
+}
+
+export const OverviewPage: React.FC = () => {
+  const { getTVLOverTime, data: tvlData } = useTVLOverTime()
+  const { getTVCOverTime, data: tvcData } = useTVCOverTime()
+
+  useEffect(() => {
+    getTVLOverTime()
+    getTVCOverTime()
+  }, [getTVLOverTime, getTVCOverTime])
+
+  const chartsData = useMemo(() => {
+    if (!tvlData || !tvcData) return
+
+    const tvcChartData: ChartDataPoint[] = []
+    const tvlChartData: ChartDataPoint[] = []
+    const riskChartData: ChartDataPoint[] = []
+
+    for (let i = 0, j = 0; i < tvlData.length && j < tvcData.length; ) {
+      const tvcDataPointDate = DateTime.fromMillis(tvcData[i].timestamp * 1000)
+      const tvlDataPointDate = DateTime.fromMillis(tvlData[j].timestamp * 1000)
+
+      let tvc = tvcData[i]
+      let tvl = tvlData[j]
+
+      if (tvcDataPointDate.day !== tvlDataPointDate.day) {
+        if (tvcDataPointDate < tvlDataPointDate) {
+          tvl = j > 0 ? tvlData[j - 1] : tvlData[j]
+          i++
+        } else {
+          tvc = i > 0 ? tvcData[i - 1] : tvcData[i]
+          j++
+        }
+      } else {
+        i++
+        j++
+      }
+
+      const timestamp = Math.min(tvc.timestamp, tvl.timestamp)
+
+      tvcChartData.push({
+        name: DateTime.fromMillis(timestamp * 1000).toFormat("M/d"),
+        value: Number(utils.formatUnits(tvc.value, 6)),
+      })
+      tvlChartData.push({
+        name: DateTime.fromMillis(timestamp * 1000).toFormat("M/d"),
+        value: Number(utils.formatUnits(tvl.value, 6)),
+      })
+      riskChartData.push({
+        name: DateTime.fromMillis(timestamp * 1000).toFormat("M/d"),
+        value: Number(utils.formatUnits(tvc.value, 6)) / Number(utils.formatUnits(tvl.value, 6)),
+      })
```

Comment:
At the moment we must traverse both datasets to compute the risk factor data, which does not allow for refactoring. I think this data can be easily computed either in the indexer or at the db leve (with something like a materialised view)

Reply:
I'll create a Trello card for it.